ipa-core/src/net/client/mod.rs

```diff
@@ -198,7 +176,8 @@ impl MpcHelperClient {
         runtime: IpaRuntime,
         client_config: &ClientConfig,
         peer_config: PeerConfig,
-        identity: ClientIdentity,
+        identity: ClientIdentity<R>,
+        header_name: &'static HeaderName,
```

Comment:
This is less readable and harder to understand in terms of allocations (because you clone it down below) than just passing it by value and cloning (HeaderName is just a wrapper around a smart pointer, so cloning is fine)

Reply:
To be sure...

You want me to give ownership (no `&`) in new `MpcHelperClient::<R>::new` and have the callers of this method e.g. `MpcHelperClient<Ring>::from_conf` clone it?

What's the readability issue? You rather see clones happening "higher-up" instead inside `new`? Let me know if you rather talk online about that part

Reply:
Possibly you can get the header name from a constant or method on `TransportRestriction` rather than storing it explicitly.

Reply:
Originally I didn't wanted to tie HTTP (net module) specific things to `TransportRestriction`. If we make it a net only parameter, I think that makes sense.

Reply:
Andy's suggestion is even better I think.